packages/platform-core/src/telemetry/usage_data_emitter.ts

```diff
@@ -0,0 +1,131 @@
+import { v4 as uuid } from 'uuid';
+import { AccountIdFetcher } from './account_id_fetcher.js';
+import { UsageData } from './usage_data.js';
+import os from 'os';
+import https from 'https';
+import { getInstallationUuid } from './get_installation_id.js';
+import { latestPayloadVersion } from './constants.js';
+import { getUrl } from './get_usage_data_url.js';
+import isCI from 'is-ci';
+import { SerializableError } from './serializable_error.js';
+
+/**
+ * Entry point for sending usage data metrics
+ */
+export class DefaultUsageDataEmitter {
+  /**
+   * Constructor for UsageDataEmitter
+   */
+  constructor(
+    private readonly libraryVersion: string,
+    private readonly sessionUuid = uuid(),
+    private readonly url = getUrl(),
+    private readonly accountIdFetcher = new AccountIdFetcher()
+  ) {}
+
+  emitSuccess = async (
+    metrics?: Record<string, number>,
```

Comment:
See https://github.com/aws-amplify/amplify-backend/pull/642#discussion_r1393493174 .

Reply:
and https://github.com/aws-amplify/amplify-backend/pull/642#discussion_r1393501107 .
above.

Reply:
The reasoning behind is that "metrics collector" should expose api to collect metrics (i.e. abstract mechanism of how data is pushed to the backend) but it should be up to caller to know what exactly is being reported.

Reply:
That's true, but we also have logic in here where we are looking for specific keys to perform certain transformations. Seems like we could get the type system to help us.

Regardless, I'm not going to block on it

Reply:
That's a good point. But I'd do that by casting these Records to something typed internally in data collector as we do this only because we want to keep old schema.